libshaderc/include/shaderc.hpp

```diff
@@ -125,6 +125,14 @@ class CompileOptions {
     shaderc_compile_options_set_generate_debug_info(options_);
   }
 
+  void SetIncluderCallbacks(
```

Comment:
:-(
The solution for (A) creates a new class, and forces a class-type based hierarchy on the clients.  I actually prefer the std::function solution.  It's more elemental and does not impose a solution structure on the client.